.gitbook/contents/hc/1_price-fetch.md

```diff
@@ -0,0 +1,60 @@
+# Price Fetch Example
+
+To dip our toes into Hybird Compute's capabilities, consider the following simple smart contract which fetches the price of different cryptocurrencies. You can view the full example in our [repository](https://github.com/bobanetwork/aa-hc-example/blob/main/contracts/contracts/TokenPrice.sol).
+
+We can start by importing some libraries (one to allow the use of Hybrid Compute, one to give exclusive access to certain functions as an "owner") and providing the general structure of our contract:
+
+```solidity
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.8.19;
+
+import "./samples/HybridAccount.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract TokenPrice is Ownable {
+    mapping(string => TokenPriceStruct) public tokenPrices;
+    HybridAccount public HA;
+
+    event FetchPriceError(uint32 err);
+    event FetchPriceRet(bytes err);
+
+    struct TokenPriceStruct {
+        string price;
+        uint256 timestamp; // Timestamp of when the price was fetched.
+    }
+
+    constructor(
+        address _hybridAccount
+    ) Ownable() {
+        HA = HybridAccount(payable(_hybridAccount));
+    }
+}
+```
+
+To this, we can add our price fetching logic:
+
+```solidity
+    function fetchPrice(string calldata token) public {
+        string memory price;
+
+        // Encode function signature, which is called on the offchain rpc server.
+        bytes memory req = abi.encodeWithSignature("getprice(string)", token);
+        bytes32 userKey = bytes32(abi.encode(msg.sender));
+        (uint32 error, bytes memory ret) = HA.CallOffchain(userKey, req);
+
+        if (error != 0) {
```

Comment:
This is correct from a documentation perspective (matches the code) but I'm not sure if the code itself makes sense. Kevin, can you check this? I'm not sure why there's an "emit" followed by a "revert", or why it's splitting the error code and 'ret' string across two different events.